src/shims/unix/fs.rs

```diff
@@ -1721,6 +1722,129 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.set_last_error(enotty)?;
         Ok(0)
     }
+
+    fn mkstemp(&mut self, template_op: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        use rand::seq::SliceRandom;
+
+        // POSIX defines the template string.
+        const TEMPFILE_TEMPLATE_STR: &str = "XXXXXX";
+        // POSIX defines the minimum number of attempts before failure.
+        // <https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/sysdeps/posix/tempname.c#L203>
```

Comment:
That link says glib uses `62**3`. Your comment says POSIX defines it, which is a much stronger claim. Do you have a source for that claim?

Reply:
Hm, I could have swore I saw it in some POSIX doc. On further inspection, this relies on `TMP_MAX` and POSIX says:

> * Minimum number of unique filenames generated by [tmpnam()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/tmpnam.html). Maximum number of times an application can call [tmpnam()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/tmpnam.html) reliably. The value of {TMP_MAX} is at least 25.
> 
> * On XSI-conformant systems, the value of {TMP_MAX} is at least 10000

https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stdio.h.html 

glbc defines `TMP_MAX` as such:

>   #ifndef TMP_MAX
>   #define TMP_MAX 238328
>   #endif

In any case, looks like the `libc` crate has `TMP_MAX` and I should probably use that:

https://docs.rs/libc/latest/libc/constant.TMP_MAX.html